location/models.py

```diff
@@ -16,6 +18,15 @@
 
 logger = logging.getLogger(__file__)
 
+def free_cache_for_user(user_id):
+    cache_name = f'location_allowed_{user_id}'
```

Comment:
Consider renaming this to `user_locations_{user_id}` to be consistent with the district cache name below.